bin/matrix

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+set -euo pipefail
+
+# Version utility for comparing versions from
+# https://stackoverflow.com/questions/4023830/how-to-compare-two-strings-in-dot-separated-version-format-in-bash
+function version { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+# Store current Ruby version
+original_ruby_version=$(cat .ruby-version)
+
+{
+  # Determine employed Ruby versions
+  readarray -t versions < <(cat .github/workflows/test.yml | grep ruby: | awk '{$1=$1};1' | sed 's/- ruby: //' | sort | uniq)
+  echo "Detected Ruby versions:"
+  for version in "${versions[@]}"
+  do
+    echo "- $version"
+  done
+  echo ""
+
+  for version in "${versions[@]}"
+  do
+    rbenv local "$version"
+
+    ruby_version=$(ruby -v)
+    rubygems_version=$(gem -v)
+    bundler_version=$(bundler -v | sed 's/Bundler version //')
+
+    echo "====================="
+    echo "Target Ruby version: $version"
+    echo ""
+    echo "Ruby: $ruby_version"
+    echo "rubygems: $rubygems_version"
+    echo "Bundler: $bundler_version"
+    echo "====================="
+    echo ""
+
+    if [ $(version $rubygems_version) -lt $(version "3.3.0") ]; then
+      echo "Please ensure that your rubygems version is > 3.3.0 for Ruby $ruby_version!"
+      echo "Install newest version:"
+      echo "gem update --system"
+      echo "Install specific version:"
+      echo "gem update --system <version>"
+      exit 1
+    fi
+
+    if [ $(version $bundler_version) -lt $(version "2.3.0") ]; then
+      echo "Please ensure that your Bundler version is > 2.3.0 for Ruby $ruby_version!"
+      echo "Install newest version:"
+      echo "gem install bundler"
+      echo "Install specific version:"
+      echo "gem install bundler:<version>"
+      exit 1
+    fi
+
+    rake matrix:install
```

Comment:
Why no `bundle exec`? Is this really stable?

Reply:
`bundle exec` breaks this rake task, because it locks the dependencies from the current Gemfile (which is only a Symlink to the Gemfile that is considered the default for testing during dev).

However, `matrix:install` and `matrix:spec` are explicitly for installing the dependencies and running the tests for multiple different Gemfiles. They are smart enough to switch versions themselves.

Reply:
I misread it, yes it just prints install commands